pkg/services/ngalert/provisioning/errors.go

```diff
@@ -44,3 +50,19 @@ func MakeErrTimeIntervalInvalid(err error) error {
 
 	return ErrTimeIntervalInvalid.Build(data)
 }
+
+func MakeErrProvenanceChangeNotAllowed(from, to models.Provenance) error {
+	if to == "" {
+		to = "none"
```

Comment:
NOTE: This is bit awkward, but better than an empty string. The model around provenance and `X-Disable-Provenance` is in need of a revamp IMO, out of scope for here.